base/reduce.jl

```diff
@@ -346,8 +349,10 @@ mapreduce_empty(::typeof(identity), op, T) = reduce_empty(op, T)
 mapreduce_empty(::typeof(abs), op, T)      = abs(reduce_empty(op, T))
 mapreduce_empty(::typeof(abs2), op, T)     = abs2(reduce_empty(op, T))
 
-mapreduce_empty(f::typeof(abs),  ::typeof(max), T) = abs(zero(T))
-mapreduce_empty(f::typeof(abs2), ::typeof(max), T) = abs2(zero(T))
+mapreduce_empty(::typeof(abs),  ::typeof(max), T) = abs(zero(T))
+mapreduce_empty(::typeof(abs),  ::typeof(min), T::Type{<:AbstractFloat}) = typemax(T)
```

Comment:
Why not support complex, integers, etc.?

```suggestion
mapreduce_empty(::typeof(abs),  ::typeof(min), T) = typemax(abs(zero(T)))
```